### PR TITLE
Refactored model test cases

### PR DESCRIPTION
**Pull request by wxwxwxwx9**
The following shows the list of changes for our tests. 

model:
(OLD) AddressBookTest.java -->  (NEW) InternshipDiaryTest.java
(OLD) ModelManagerTest.java -->  (REFACTORED) ModelManagerTest.java
(OLD) UserPrefsTest.java -->  (REFACTORED) UserPrefsTest.java

testutil: 
(OLD) AddressBookBuilder.java -->  (NEW) InternshipDiaryBuilder.java
(OLD) PersonBuilder.java --> (NEW) InternshipApplicationBuilder.java
(OLD) TypicalPersons.java --> (NEW) TypicalInternshipApplications.java

I have left the (OLD) files in the code base for now, until we have verified that it is safe/wise to delete them. 

After this is settled and approved, I will proceed to create more tests for new classes like: 
- Role.java
- Priority.java
- InternshipApplication.java

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -18,12 +18,14 @@
  * Represents the in-memory model of the internship diary data.
  */
 public class ModelManager implements Model {
+
```
Comment (foochifa) [inline src/main/java/seedu/address/model/ModelManager.java:21]: Thanks for making the refactor more readable! Sorry I missed out the spaces

File: src/test/java/seedu/address/model/InternshipDiaryTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Test
+    public void resetData_withValidReadOnlyInternshipDiary_replacesData() {
+        InternshipDiary newData = getTypicalInternshipDiary();
+        internshipDiary.resetData(newData);
+        assertEquals(newData, internshipDiary);
+    }
+
+    @Test
+    public void resetData_withDuplicateInternship_throwsDuplicateInternshipException() {
```
Comment (foochifa) [inline src/test/java/seedu/address/model/InternshipDiaryTest.java:47]: I think this checks to see if resetData properly throws exception when there are two similar internship application

**Review by foochifa — COMMENTED** [review]
Goodjob in the refactoring and the new test classes! Thanks!

File: src/test/java/seedu/address/testutil/InternshipApplicationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+public class InternshipApplicationBuilder {
+
+    public static final String DEFAULT_COMPANY = "Google";
+    public static final String DEFAULT_ROLE = "Product Manager";
+    public static final String DEFAULT_ADDRESS = "1600 Amphitheatre Parkway";
+    public static final String DEFAULT_PHONE = "99999999";
+    public static final String DEFAULT_EMAIL = "richardma@gmail.com";
+    public static final Date DEFAULT_APPLICATION_DATE = new Date(2020, 01, 01);
+    public static final Integer DEFAULT_PRIORITY = 10;
+    public static final Status DEFAULT_STATUS = Status.APPLICATION_DONE;
```
Comment (gerhean) [inline src/test/java/seedu/address/testutil/InternshipApplicationBuilder.java:26]: Subarashi example!

**Pull request merged**